tests/test_branch_stats.py

```diff
@@ -197,6 +288,12 @@ def assertListAlmostEqual(self, x, y):
         for a, b in zip(x, y):
```

Comment:
I should have spotted this before, but can we add a 
```
self.assertEqual(len(x), len(y))
```
here please? This will currently happily compare [2], [2, 3, 4, 5] as equal.